src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -57,19 +57,16 @@ class LogInWithShortLivedTokenPage extends Component {
         const email = lodashGet(this.props.route.params, 'email', '');
         const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
 
-        // User is trying to transition with a different account than the one they are currently signed in as so we will sign out and then sign in with the new authToken
-        if (email !== this.props.session.email) {
-            Session.signOut();
+        const signedIn = this.props.session && this.props.session.authToken;
+        if (!signedIn) {
             Session.signInWithShortLivedToken(accountID, email, shortLivedToken);
             return;
         }
 
-        // If the user is revisiting the component authenticated with the right account, we don't need to do anything, the componentWillUpdate when betas are loaded and redirect
-        if (this.props.session.authToken && email === this.props.session.email) {
-            return;
+        // User is trying to transition with a different account than the one they are currently signed in as so we will sign out and then sign in with the new authToken
```

Comment:
NAB: Could be helpful to add that they will be signed in because they will be navigated back to this component as a public screen. It could be confusing for someone reading this in the future to figure out how this is happening.

Reply:
Good idea, I updated the comment.